### PR TITLE
[Enhancement] Support to recognize skip.header.line.count in hive's textfile

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## Why I'm doing:
Support `skip.header.line.count` tblproperties in hive table.

demo:
```sql
create table csv_big_skip
(c1 string, c2 int) stored as textfile tblproperties("skip.header.line.count"="10");
```

## What I'm doing:

Support it

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/hdfs_scanner_text.cpp [hunk 1]
```diff
@@ -225,29 +227,48 @@ Status HdfsTextScanner::do_init(RuntimeState* runtime_state, const HdfsScannerPa
...
+}
 
+Status HdfsTextScanner::_setup_compression_type(const TTextFileDesc& text_file_desc) {
     // by default it's unknown compression. we will synthesise informaiton from FE and BE(file extension)
     // parse compression type from FE first.
-    _compression_type = CompressionTypePB::UNKNOWN_COMPRESSION;
+    CompressionTypePB compression_type;
     if (text_file_desc.__isset.compression_type) {
-        _compression_type = CompressionUtils::to_compression_pb(text_file_desc.compression_type);
+        compression_type = CompressionUtils::to_compression_pb(text_file_desc.compression_type);
```
Comment (DorianZheng) [inline be/src/exec/hdfs_scanner_text.cpp:238]: 👍

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java [hunk 2]
```diff
@@ -584,28 +590,29 @@ public static TextFileFormatDesc toTextFileFormatDesc(Map<String, String> serdeP
...
             // Support for hive org.apache.hadoop.hive.serde2.OpenCSVSerde
             // https://cwiki.apache.org/confluence/display/hive/csv+serde
-            fieldDelim = serdeParams.getOrDefault("separatorChar", "");
+            fieldDelim = parameters.getOrDefault(OpenCSVSerde.SEPARATORCHAR, "");
         }
-        String lineDelim = serdeParams.getOrDefault("line.delim", "");
-        String mapkeyDelim = serdeParams.getOrDefault("mapkey.delim", "");
+        String lineDelim = parameters.getOrDefault(serdeConstants.LINE_DELIM, "");
+        String mapkeyDelim = parameters.getOrDefault(serdeConstants.MAPKEY_DELIM, "");
+        int skipHeaderLineCount = Integer.parseInt(parameters.getOrDefault(serdeConstants.HEADER_COUNT, "0"));
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java:607]: better check if it's legit, such as >= 0

File: be/src/exec/hdfs_scanner_text.h [hunk 3]
```diff
@@ -44,19 +44,22 @@ class HdfsTextScanner final : public HdfsScanner {
...
-    Status parse_csv(int chunk_size, ChunkPtr* chunk);
     int64_t estimated_mem_usage() const override;
 
 private:
-    // create a reader or re init reader
-    Status _create_or_reinit_reader();
+    Status _create_csv_reader();
+    Status _setup_compression_type(const TTextFileDesc& text_file_desc);
+    Status _setup_delimiter(const TTextFileDesc& text_file_desc);
+    StatusOr<bool> _has_uft8_bom() const;
```
Comment (GavinMar) [inline be/src/exec/hdfs_scanner_text.h:53]: typo for `utf8`?

File: fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java (hunk 2 above)
Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetastoreApiConverter.java:607]: done

File: be/src/exec/hdfs_scanner_text.h (hunk 3 above)
Comment (Smith-Cruise) [reply be/src/exec/hdfs_scanner_text.h]: done

**Review by GavinMar — APPROVED** [review]

**Review by DorianZheng — APPROVED** [review]

File: be/src/exec/hdfs_scanner.cpp [hunk 4]
```diff
@@ -627,4 +628,12 @@ void HdfsScanner::move_split_tasks(std::vector<pipeline::ScanSplitContextPtr>* s
     }
 }
 
+CompressionTypePB HdfsScanner::get_compression_type_from_path(const std::string& filename) {
```
Comment (trueeyu) [inline be/src/exec/hdfs_scanner.cpp:631]: Duplicate with the function return_compression_type_from_filename of hdf_scanner_text.cpp ?

Make it to one common function?

File: gensrc/thrift/Descriptors.thrift [hunk 5]
```diff
@@ -100,6 +100,8 @@ struct TTextFileDesc {
     
     // escape character
     8: optional i8 escape
+
+    9: optional i32 skip_header_line_count
```
Comment (trueeyu) [inline gensrc/thrift/Descriptors.thrift:104]: After adding this Skip, will it cause problems with the optimizer, such as Limit 1. In the past, FE would select 1 file, but with this skip, problems may occur?

**Review by trueeyu — APPROVED** [review]

**Pull request merged**